src/amo/components/SearchForm.js

```diff
@@ -39,7 +45,17 @@ export class SearchFormBase extends React.Component {
   }
 
   render() {
-    const { api, i18n, pathname, query } = this.props;
+    const { addonType, api, i18n, pathname, query } = this.props;
+
+    let placeholderText;
+    if (addonType === ADDON_TYPE_EXTENSION) {
```

Comment:
Again, we already used `addonType` in other search bits so I'd rather let search things refer to variables used for searching; the current view context is a bit more abstracted from search.